torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -204,35 +204,45 @@ at::DimVector inversePermutation(
   }
 }
 
+template <typename T, typename CONTAINER>
+void encode_buf(T value, CONTAINER& buffer) {
```

Comment:
1. naming: encodeBuff (or even better, `encodeBuffer` since we don't need to save 2 keystrokes here)

2. why does it have to be a template?

Reply:
1. oops. :stuck_out_tongue: 
2. I was playing with different container & types. I can get rid of the template as it's not necessary any more.